src/bundle.c

```diff
@@ -620,6 +620,42 @@ static gboolean encrypt_bundle_payload(const gchar *bundlepath, RaucManifest *ma
 	return TRUE;
 }
 
+static gboolean decrypt_bundle_payload(RaucBundle *bundle, RaucManifest *manifest, GError **error)
+{
+	gboolean res = FALSE;
+	g_autofree guint8 *key = NULL;
+	g_autofree gchar* decpath = NULL;
+	g_autoptr(GFile) decgfile = NULL;
+
+	g_return_val_if_fail(bundle, FALSE);
+	g_return_val_if_fail(manifest, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	decpath = g_strconcat(bundle->path, ".dec", NULL);
```

Comment:
Use a random filename here.

Reply:
I'll create a tmpdir here as we do it in convert already.